File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def func(n):
    to3_list = []
    answer = 0
    while n != 0:
        r = n % 3
        to3_list.append(r)
        n //= 3
    
    for i in range(len(to3_list)):
        if to3_list[i] == 0:
            continue
        else:
            answer += to3_list[i] * (3 ** (len(to3_list) - (i + 1)))
    return answer

def solution(n):
    return func(n)

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.js
function solution(n) {
    let answer = 0;
    let arr =[];
    
    while(n !== 0){
        arr.push(n % 3);
        n = Math.floor(n / 3);
    }
    for(let i=0; i<arr.length; i++) {
        answer += arr[i] * Math.pow(3, arr.length - i - 1);
    }
    return answer;
}